### PR TITLE
 feat: update template to use workers-rs project and include example

### PR DESCRIPTION
**Pull request by nilslice**
- [x] convert repo to use `worker` crate from the [`workers-rs`](https://github.com/cloudflare/workers-rs) project.
- [x] update generated example to use the new crate.
- [x] update README to provide high-level instructions for using the repo.
- [x] fix the templatized wrangler.toml and other files to use the right placeholders.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,22 +1,54 @@
...
 
-#[wasm_bindgen]
-pub fn greet() -> String {
-    "Hello, wasm-worker!".to_string()
+#[event(fetch)]
+pub async fn main(req: Request, env: Env) -> Result<Response> {
+    log_request(&req);
+
+    // Optionally, get more helpful error messages written to the console in the case of a panic.
+    utils::set_panic_hook();
```
Comment (Electroid) [inline src/lib.rs:21]: This is just a side note, not really something that needs to be addressed here. But does this have a significant performance penalty? If not, could this just be baked in, automatically?

**Review by Electroid — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (nilslice) [reply src/lib.rs:21]: @Electroid I'm not sure about the exact performance impact, but there is definitely _some_ impact. 

I personally think it improves the dev experience enough to justify it baked in and make it possible to opt-out via features.

Comment (nilslice) [reply src/lib.rs:21]: I'll merge this in as-is, and get the utils::set_panic_hook added in with an opt-out tonight or tomorrow.. just want to test that this template works as expected

File: README.md [hunk 2]
```diff
@@ -1,39 +1,41 @@
...
+Read the latest `worker` crate documentation here: https://docs.rs/worker
 
-```
-wasm-pack build
-```
+## WebAssembly
 
-### 🔬 Test in Headless Browsers with `wasm-pack test`
+`workers-rs` (the Rust SDK for Cloudflare Workers used in this template) is meant to be executed as 
+compiled WebAssembly, and as such so **must** all the code you write and depend upon. All crates and
```
Comment (threepointone) [inline README.md:32]: This sentence is odd, did you miss a word here? "and as such so **must** all the code you write and depend upon."

Comment (nilslice) [reply README.md:32]: haha no I don't think so - but it is a little wordy to just say "all your code and dependencies have to compile to wasm"

Comment (threepointone) [reply README.md:32]: Oh ok I understand, lgtm 👍

**Pull request merged**